### PR TITLE
fix: Use launch configuration to resolve the classpath

### LINKED ISSUE #623: Multi-Root Workspace with two Maven projects uses wrong pom dependencies
VS Code 1.31.1 - Java Test Runner 0.14.0 - Maven for Java 0.14.2

I have two Maven Projects, each with UnitTests. If I open them individually all Tests run without problems. If I open them in a multi-root workspace and run Tests of the second Project, it seems to try and run the Tests with the dependencies of the first Project. The Projects use different Spring versions and the Testrunner tries to run the Tests of the second project using the Spring-Version from the first Project.

What is expected:
Running Tests in a Multi-Root Workspace project with several Maven Projects should use the pom Dependencies of the Project where the Test is located just as if the Root was Opened alone as an individual project.

### PR DESCRIPTION
**Pull request by jdneo**
Fix #623

### CODE REVIEW

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java [hunk 1]
```diff
@@ -76,4 +93,87 @@
...
+                        entry.getClasspathProperty() == IRuntimeClasspathEntry.CLASS_PATH) {
+                    classpaths.add(location);
+                }
+                // TODO: support module path
+            }
+        }
+        return classpaths.toArray(new String[classpaths.size()]);
+    }
+
+    private static class JavaApplicationLaunchConfiguration extends LaunchConfiguration {
```
Comment (testforstephen) [inline java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java]: JunitLaunchConfiguration?

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java [hunk 2]
```diff
@@ -76,4 +93,87 @@
...
+        public static final String JAVA_APPLICATION_LAUNCH = "<?xml version=\"1.0\" encoding=\"UTF-8\"" +
```
Comment (testforstephen) [inline java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java]: JUNIT_LAUNCH

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java [hunk 3]
```diff
@@ -76,4 +93,87 @@
...
+
+        protected JavaApplicationLaunchConfiguration(IProject project) throws CoreException {
+            super(String.valueOf(new Date().getTime()), null, false);
+            this.project = project;
+            if (ProjectUtils.isMavenProject(project)) {
+                classpathProvider = "org.eclipse.m2e.launchconfig.classpathProvider";
+                sourcepathProvider = "org.eclipse.m2e.launchconfig.sourcepathProvider";
+            } else if (ProjectUtils.isGradleProject(project)) {
+                classpathProvider = "org.eclipse.buildship.core.classpathprovider";
+            }
```
Comment (Eskibear) [inline java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java]: what if not maven or gradle?

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java [hunk 4]
```diff
@@ -76,4 +93,87 @@
         }
         return classPathList.toArray(new String[classPathList.size()]);
     }
+
+    private static String[] resolveClasspath(IJavaProject javaProject) throws CoreException {
+//        final JUnitLaunchConfigurationDelegate delegate = new JUnitLaunchConfigurationDelegate();
```
Comment (Eskibear) [inline java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java]: unused comments here?

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java (hunk 1 above)
Comment (jdneo) [reply java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java]: Updated

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java (hunk 2 above)
Comment (jdneo) [reply java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java]: Updated

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java (hunk 4 above)
Comment (jdneo) [reply java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java]: Updated

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java (hunk 3 above)
Comment (jdneo) [reply java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java]: Works as expected.

demo

**Pull request merged**